kernel/thread.c

```diff
@@ -577,8 +577,8 @@ void _k_thread_single_abort(struct k_thread *thread)
 	/* Clear initailized state so that this thread object may be re-used
 	 * and triggers errors if API calls are made on it from user threads
 	 */
-	_k_object_uninit(thread);
 	_k_object_uninit(thread->stack_obj);
+	_k_object_uninit(thread);
```

Comment:
why are you changing the order?
these aren't mutexes and all this happens with interrupts locked

Reply:
i felt it's logical to uninit the object in the dependency order like constructor/destructor. Here in this case we can uninit the thread's stack_obj first and then the thread itself.

Reply:
andrew, if it does not make sense i can remove the change and resubmit.

Reply:
still don't feel like this is quite an appropriate analogy since k_object_uninit isn't a destructor but the change is harmless